src/test/java/net/greatstart/controllers/UserRestControllerTest.java

```diff
@@ -0,0 +1,135 @@
+package net.greatstart.controllers;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import net.greatstart.dto.DtoUserProfile;
+import net.greatstart.mappers.UserProfileMapper;
+import net.greatstart.model.User;
+import net.greatstart.services.UserService;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.setup.MockMvcBuilders.standaloneSetup;
+
+import java.security.Principal;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(MockitoJUnitRunner.class)
+public class UserRestControllerTest {
+
+    private static final long ID = 1L;
+    private static final String TEST_USER_PROFILE = "/api/user/" + ID;
+
+
+    @Mock
+    private Principal principal;
+    @Mock
+    private UserService userService;
+    @Mock
+    private UserProfileMapper userMapper;
+
+    @InjectMocks
+    private UserRestController controller;
+    private MockMvc mvc;
+    private User user;
+    private DtoUserProfile dtoUser;
+
+
+    @Before
+    public void setUp() {
+        user = new User();
```

Comment:
You don't need @Before section to do this, you can do this inline above where you declare a variable